### DISCUSSION TITLE
Data Importer imports in the future

### DISCUSSION THREAD

**Discussion by cprpolo** [firefly-iii/firefly-iii#9332] [Firefly III] [answered]
### Support guidelines

- [x] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [ ] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I'm trying to import transactions from a csv file and some/most of the imported appeared with the Transaction Date in the future

here is an example
![Image](https://github.com/user-attachments/assets/d9957e62-ab5a-4e24-b8be-a832f30e4069)

![Image](https://github.com/user-attachments/assets/27a3f6ea-bc25-40dc-b381-a4f50cd68e41)

The date format in the csv:
![Image](https://github.com/user-attachments/assets/07fe574e-7598-4160-acda-9059ff3eaf7d)

and therefore i think my date format is correct in the configuration: n/j/Y
Config file:

{
    "version": 3,
    "source": "ff3-importer-1.5.6",
    "created_at": "2024-10-07T10:06:47+01:00",
    "date": "n\/j\/Y",
    "default_account": 1,
    "delimiter": "comma",
    "headers": true,
    "rules": true,
    "skip_form": false,
    "add_import_tag": true,
    "roles": [
        "account-number",
        "date_transaction",
        "description",
        "note",
        "_ignore",
        "amount_debit",
        "amount_credit",
        "_ignore",
        "currency-code",
        "_ignore",
        "_ignore",
        "_ignore"
    ],
    "do_mapping": [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "mapping": [],
    "duplicate_detection_method": "none",
    "ignore_duplicate_lines": false,
    "unique_column_index": 0,
    "unique_column_type": "internal_reference",
    "flow": "file",
    "content_type": "csv",
    "custom_tag": "AIB Import",
    "identifier": "0",
    "connection": "0",
    "ignore_spectre_categories": false,
    "grouped_transaction_handling": "",
    "use_entire_opposing_address": false,
    "map_all_data": true,
    "accounts": [],
    "date_range": "",
    "date_range_number": 30,
    "date_range_unit": "d",
    "date_not_before": "",
    "date_not_after": "",
    "nordigen_country": "",
    "nordigen_bank": "",
    "nordigen_requisitions": [],
    "nordigen_max_days": "90",
    "conversion": false,
    "ignore_duplicate_transactions": false
}
It may be a stupid mistake but i am trying few days and can't figure it out.
Sorry if it is a stupid question

Any help appreciate it

### Debug information

Let me know what you need

### Expected behaviour

To import with the correct date of the transaction

### Steps to reproduce

1.    Open Data importer
2.   Choose the csv file
3.    Choose the config file (or configure it manually with the wizard)
4.    Start the job
5.    Wait until it completes

**Comment by JC5** [comment]
It looks correct. Could you share a few lines from the file directly?

If you want, send them to james@firefly-iii.org.

**Comment by cprpolo** [comment]
I just emailed you there :)

**Comment by JC5** [comment, marked as answer]
Found the issue. Although Excel displays the dates as you show them in the screenshot, the actual format _inside the file_ is as follows:

`20/09/2024`

From that, it follows that the correct format for the CSV importer is `d/m/Y`.

Let me know!

**Reply by cprpolo** [reply]
That was it!
Thank you for your help @JC5
